platform/security/sts/security-sts-propertyclaimshandler/src/main/java/org/codice/ddf/security/sts/claims/property/AttributeFileClaimsHandler.java

```diff
@@ -148,16 +130,51 @@ public ProcessedClaimCollection retrieveClaimValues(
   }
 
   @Override
-  public List<String> getSupportedRealms() {
-    return supportedRealms;
+  public List<Serializable> getValues(final String attributeName) {
+    return Optional.ofNullable(systemHighAttributes.get(attributeName))
+        .orElse(Collections.emptyList());
   }
 
-  @Override
-  public String getHandlerRealm() {
-    return realm;
+  public void setAttributeFileLocation(String attributeFileLocation) throws IOException {
+    this.attributeFileLocation =
+        notNull(attributeFileLocation, "attributeFileLocation may not be null");
+    init();
   }
 
-  public void setAttributeFileLocation(String attributeFileLocation) {
-    this.attributeFileLocation = attributeFileLocation;
+  private void setSupportedClaimTypes() {
+    Set<Map.Entry<String, Object>> entries = json.entrySet();
+    Set<String> claims = new HashSet<>();
+    for (Map.Entry<String, Object> entry : entries) {
+      Object value = entry.getValue();
+      if (value instanceof Map) {
+        Set keySet = ((Map) value).keySet();
+        for (Object key : keySet) {
+          claims.add((String) key);
+        }
+      }
+    }
+
+    supportedClaimTypes = claims.stream().map(URI::create).collect(Collectors.toList());
+  }
+
+  private void setSystemHighAttributes() {
+    final Map<String, List<Serializable>> newSystemHighAttributes = new HashMap<>();
+
+    final Map<String, Object> systemHighUser =
+        (Map<String, Object>) json.get(SystemBaseUrl.getHost());
+    for (Map.Entry<String, Object> systemHighAttribute : systemHighUser.entrySet()) {
+      final String systemHighAttributeName = systemHighAttribute.getKey();
+
+      final Object systemHighAttributeValue = systemHighAttribute.getValue();
+      if (systemHighAttributeValue instanceof String) {
+        newSystemHighAttributes.put(
+            systemHighAttributeName, Collections.singletonList((String) systemHighAttributeValue));
+      } else {
+        newSystemHighAttributes.put(
+            systemHighAttributeName, (List<Serializable>) systemHighAttribute.getValue());
+      }
+    }
+
+    this.systemHighAttributes = newSystemHighAttributes;
```

Comment:
❗️ `systemHighAttributes ` needs to be `volatile` to make sure it's not being changed while `getValues()` is called.